internal/scheduler/additional_verification_test.go

```diff
@@ -338,3 +339,102 @@ func TestSchedulerMonitorLoop(t *testing.T) {
 	require.NoError(err)
 	assert.Equal(string(job.Interrupted), run.Status)
 }
+
+func TestSchedulerFinalStatusUpdate(t *testing.T) {
```

Comment:
Did this come in from merging in from `main` or something? Or is this relevant to the current PR in a way I'm not seeing?

It's not a problem if it's the former, but _if so_ you may want to get in the habit of doing a `main` -> `base long-term branch` -> `feature branch` merge instead of merging directly from `main` to `feature branch` to keep PRs clean.

Reply:
Yes this is from merging `main`. I will cleanup this PR to follow that pattern of `main` -> `long-term` to clean this up.

Reply:
cleaned the PR up